frontend/public/components/secrets/create-secret.tsx

```diff
@@ -144,6 +163,308 @@ const withSecretForm = (SubForm) => class SecretFormComponent extends React.Comp
   }
 };
 
+const getImageSecretKey = (secretType) => {
```

Comment:
Hmm... both of these functions `getImageSecretKey` and  `getImageSecretType` have just two keys or types to return, since they are used for the image-secret types, which are just 'kubernetes.io/dockerconfigjson and 'kubernetes.io/dockercfg', so I this that simple `if` statement with defaulting to the other type/key should be sufficient.

Reply:
Or do we want to default to `Opaque` SecretType ? So we have control over the returned types ?

Reply:
I think the implementation is fine, just that we should add the proper types to the function signature to take advantage of TypeScript